blueprints/networking/f5-lb/main.tf

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  # F5 configs
+  as3_url  = "https://github.com/F5Networks/f5-appsvcs-extension/releases/download/v3.46.0/f5-appsvcs-3.46.0-5.noarch.rpm"
+  as3_ver  = format("v%s", split("-", split("/", local.as3_url)[length(split("/", local.as3_url)) - 1])[2])
+  cfe_ver  = format("v%s", split("-", split("/", local.cfe_url)[length(split("/", local.cfe_url)) - 1])[3])
+  cfe_url  = "https://github.com/F5Networks/f5-cloud-failover-extension/releases/download/v1.15.0/f5-cloud-failover-1.15.0-0.noarch.rpm"
+  do_url   = "https://github.com/F5Networks/f5-declarative-onboarding/releases/download/v1.39.0/f5-declarative-onboarding-1.39.0-4.noarch.rpm"
+  do_ver   = format("v%s", split("-", split("/", local.do_url)[length(split("/", local.do_url)) - 1])[3])
+  fast_url = "https://github.com/F5Networks/f5-appsvcs-templates/releases/download/v1.25.0/f5-appsvcs-templates-1.25.0-1.noarch.rpm"
+  fast_ver = format("v%s", split("-", split("/", local.fast_url)[length(split("/", local.fast_url)) - 1])[3])
+  init_url = "https://cdn.f5.com/product/cloudsolutions/f5-bigip-runtime-init/v1.6.2/dist/f5-bigip-runtime-init-1.6.2-1.gz.run"
+  ts_ver   = format("v%s", split("-", split("/", local.ts_url)[length(split("/", local.ts_url)) - 1])[2])
+  ts_url   = "https://github.com/F5Networks/f5-telemetry-streaming/releases/download/v1.33.0/f5-telemetry-1.33.0-1.noarch.rpm"
+}
+
+resource "google_compute_address" "dataplane_ip_addresses" {
+  for_each = toset(var.zones)
+  name     = "${var.prefix}-${var.name}-f5-ip-dataplane-${each.key}"
+  address = (
+    var.f5_ip_addresses_config[each.key]["dataplane"] == null
+    ? null
+    : var.f5_ip_addresses_config[each.key]["dataplane"]
+  )
+  address_type = "INTERNAL"
+  project      = var.project_id
+  region       = var.region
+  subnetwork   = var.network_self_links["dataplane"]["subnetwork"]
+}
+
+resource "google_compute_address" "mgmt_ip_addresses" {
+  for_each     = toset(var.zones)
+  name         = "${var.prefix}-${var.name}-f5-ip-mgmt-${each.key}"
+  project      = var.project_id
+  address_type = "INTERNAL"
+  region       = var.region
+  subnetwork   = var.network_self_links["mgmt"]["subnetwork"]
+
+  address = (
+    var.f5_ip_addresses_config[each.key]["mgmt"] == null
+    ? null
+    : var.f5_ip_addresses_config[each.key]["mgmt"]
+  )
+}
+
+module "bigip-vms" {
```

Comment:
you can have this module create instance groups so you avoid the two resources for groups